psoc/grobot_workspace/grobot_module_controller.cydsn/dht11.c

```diff
@@ -0,0 +1,234 @@
+//#include <project.h>
+
+#include "dht11.h"
+#include <stdint.h>
+#include <dht.h>
+
+#define null 0
+
+void delay(void);
+void unpack_dht_data(void);
+
+uint8_t check;
+uint8_t decode_flag;
+uint8_t decoded_temperature_data;
+uint8_t decoded_humidity_data;
+
+int temperature_values[8];
+int humidity_values[8];
+int diff_count_values[44]={0};
+int countvalues_datatimer[44]={0};
+int executed;
+int count;
+int risingedge_counts=0;
+
+//_CY_ISR_PROTO(Delay_Timer_ISR_Handler);
+//_CY_ISR_PROTO(DHT_Pin_ISR_Handler);
+
+//What Return_Decoded_Temperature does:
+//    -Returns the data in decoded_temperature_data
+// 
+//Parameters: None
+//Returns: uint8_t decoded_temperature_data
+
+uint8_t dht_get_temperature(){
+    
+    if(decoded_temperature_data == null){
+        return -1;
+    }
+    else{
+        return decoded_temperature_data; 
+    }    
+}    
+
+//What Return_Decoded_Humidity does:
+//    -Returns the data in decoded_humidity_data
+// 
+//Parameters: None
+//Returns: uint8_t decoded_humidity_data
+
+uint8_t dht_get_humidity(){
+    
+    if(decoded_humidity_data == null){
+        return -1;
+    }else{
+        return decoded_humidity_data;
+    }    
+}   
+
+
+//What delay_funct does:
+//    - Enable Delay_Timer_ISR to detect Delay_Tiemr Terminal Count interrupt
+//    - Start the Delay_Timer component.
+//    - This function does not use the CPU delay API, CyDelay(), but uses Delay_Timer to generate delay. 
+//
+//Parameters: None
+//Return: None
+
+void delay_funct(void){
+    count = 1;
+    executed = 0;
+    // Enable Delay_timer_ISR
```

Comment:
You don't have to be *that* verbose. I would assume that a function called ```Delay_Timer_ISR_Enable``` enables the ISR. In general, I would avoid comments that don't tell us something which is not obvious.